src/sql/workbench/services/notebook/browser/models/notebookModel.ts

```diff
@@ -1317,6 +1317,9 @@ export class NotebookModel extends Disposable implements INotebookModel {
 							this._onValidConnectionSelected.fire(false);
 						}
 					});
+				this.adstelemetryService.createActionEvent(TelemetryKeys.TelemetryView.Notebook, TelemetryKeys.NbTelemetryAction.ConnectionChanged)
```

Comment:
We should keep using the `sendNotebookTelemetryActionEvent` for this since that adds additional metadata we want for all Notebooks (right now it's just the guid which is less important for us, but still something we should be keeping when possible). 

I'd suggest extending that to optionally take in a connection info - and if one is provided then call `withConnectionInfo` on the event there.